### ISSUE TITLE
[JS] Class static block available by default

### ISSUE THREAD

**Issue by Rumyra** [mdn/content#8606]
## Acceptance Criteria

- [x] The listed features are documented sufficiently on MDN
- [x] BCD is updated
- [x] Interactive example and data repos are updated if appropriate
- [ ] The content has been reviewed as needed

### For folks helping with Firefox related documentation

- [x] Set bugs to `dev-doc-complete`
- [x] Add entry to Firefox release notes if feature is enabled in release
or
- [x] Add entry to Firefox experimental features page if feature is not yet enabled in release

## Features to document

Class static blocks - see https://github.com/tc39/proposal-class-static-block which is helpful. Check classes & static docs for updates

## Related Gecko bugs

https://bugzilla.mozilla.org/show_bug.cgi?id=1725689

**Comment by hamishwillee** [comment]
@Rumyra, JavaScript is mad. I'll at least start this, but will need your review.

So I _think_ this is worth having its own topic under classes - [Classes > Class_static_initialization_blocks](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Classes/Class_static_initialization_blocks). Probably also updates to [Classes > static](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static) and maybe [Classes >Static methods and properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#static_methods_and_properties). 
Might need to be mentions in [Classes > Public_class_fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields#public_static_fields) (not so sure on that).

Some really useful resources, particularly the first one:
- https://2ality.com/2021/09/class-static-block.html - good worked example
- V8 explainer: https://v8.dev/features/class-static-initializer-blocks
- Original proposal: https://github.com/tc39/proposal-class-static-block
- v8 release note: https://v8.dev/blog/v8-release-94 (planned for Chrome 94)

Status:
- [ ] BCD: https://github.com/mdn/browser-compat-data/pull/12508
- [ ] Release note
- [ ] Docs
- [ ] Dev-doc-complete

**Comment by Rumyra** [comment]
> @Rumyra, JavaScript is mad.

Yup /me nods wildly

I thought this might be a bit of work. I'll come back to it first thing on Monday to check through the above properly.

Happy for you to get started and I'll try and support as best I can 👍

**Comment by hamishwillee** [comment]
@Rumyra Great. I did a first draft in https://github.com/mdn/content/pull/9281 - I am kind of hoping that if you don't like that you will take the job over. I spent a lot of time on this today. I do NOT think it is hard - I think I'm just not particularly experience in JavaScript and the time remaining for this cycle is short.

I decided that https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static would be a good place for this, but it would end up hiding the 80% people need most. So in the end I created a new doc for this and linked.
